app/src/main/java/com/parasoft/demoapp/component/OrderDialog.java

```diff
@@ -100,6 +101,9 @@ public void onResponse(@NonNull Call<ResultResponse<OrderResponse>> call, @NonNu
                         orderInfo = response.body().getData();
                         if (!orderInfo.getReviewedByAPV()) {
                             updateOrderStatus(orderInfo);
```

Comment:
I think the action to remove "New" should only happen when the API call to update status succeed, which is inside "onResponse" of "updateOrderStatus" method. 
Also I don't see the value of a map because we only need to remove "New" for designated order and will never set it back. So please check to see whether it's possible that you can get the specific TextView for "New" and pass it in OrderDialog and set the visibility when the call succeed.
Maybe you can use "recyclerView.getLayoutManager().getChildAt()" to get the child view in recycler and then use findViewById() to get the TextView for new status.